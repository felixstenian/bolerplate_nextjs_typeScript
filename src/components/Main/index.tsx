import * as Style from './styles'

const Main = () => {
  return (
    <Style.Wrapper>
      <Style.Logo
        src="/img/logo.svg"
        alt="Imagem de um átomo e react avançado escrito ao lado."
      />
      <Style.Title>React Avançado</Style.Title>
      <Style.Description>
        TypeScript, ReactJS, NextJS e Styled Components
      </Style.Description>
      <Style.Illustration
        src="/img/hero-illustration.svg"
        alt="Desenvolvedor de frete para uam tela com código"
      />
    </Style.Wrapper>
  )
}

export default Main
