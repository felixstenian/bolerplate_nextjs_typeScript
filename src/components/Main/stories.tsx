import { withKnobs, text } from '@storybook/addon-knobs'
import { Meta, Story } from '@storybook/react'
import Main from '.'

export default {
  title: 'Main',
  component: Main,
  decorators: [withKnobs]
} as Meta

export const Basic: Story = () => (
  <Main title={text('Title', 'React AvanÃ§ado StoryBook')} />
)
