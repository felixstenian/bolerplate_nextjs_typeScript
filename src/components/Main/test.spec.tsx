import { render, screen } from '@testing-library/react'

import Main from '.'

describe('<Main />', () => {
  it('1. should render the reading', () => {
    const { container } = render(<Main />)

    expect(
      screen.getByRole('heading', { name: /react avan√ßado/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
