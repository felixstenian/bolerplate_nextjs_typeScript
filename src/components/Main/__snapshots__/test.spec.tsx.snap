// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> 1. should render the reading 1`] = `
<main>
  <h2>
    React Avan√ßado
  </h2>
</main>
`;
